src/conn/conn_sweep.c

```diff
@@ -266,13 +282,20 @@ __sweep_server(void *arg)
 	/*
 	 * Sweep for dead and excess handles.
 	 */
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(conn, WT_CONN_SERVER_SWEEP)) {
-		/* Wait until the next event. */
-		__wt_cond_wait(session,
-		    conn->sweep_cond, conn->sweep_interval * WT_MILLION);
+	for (;;) {
+		if (!__sweep_server_run_chk(session))
+			break;
+
 		__wt_seconds(session, &now);
```

Comment:
I just realized you moved the call to `wt_seconds` to before the `cond_wait`.  It originally was after the wait.  Was that change intentional?